packages-tests/BetterPhpDocParser/PhpDocInlineHtml/config/configured_rule.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+use Rector\Core\Configuration\Option;
+use ViewScopeRector\Inferer\Rocket\TestFileLocator;
+
+return static function (ContainerConfigurator $containerConfigurator): void {
+    $services = $containerConfigurator->services();
+    $parameters = $containerConfigurator->parameters();
+
+    require_once __DIR__.'/../Source/InlineHtmlRector.php';
```

Comment:
I feel like we are getting closer ;).
as of now the rector test looks nicer and it most of the time does what I expect.

I dont like this manual autoloading here.. I added it because I think rector does not expect to find a rector-impl within a tests `Source` folder and therefore the rector class will not be found.. it works sometimes... sometimes not.

```
$ vendor/bin/phpunit packages-tests/BetterPhpDocParser/PhpDocInlineHtml/TestInlineHtmlTest.php --debug
PHPUnit 9.5.4 by Sebastian Bergmann and contributors.

Test 'Rector\Tests\BetterPhpDocParser\PhpDocInlineHtml\TestInlineHtmlTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))' started
Test 'Rector\Tests\BetterPhpDocParser\PhpDocInlineHtml\TestInlineHtmlTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))' ended


Time: 00:01.002, Memory: 32.00 MB

There was 1 error:

1) Rector\Tests\BetterPhpDocParser\PhpDocInlineHtml\TestInlineHtmlTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
Error: Class "Rector\Tests\BetterPhpDocParser\PhpDocInlineHtml\InlineHtmlRector" not found

C:\xampp7.3\htdocs\rector-src\tmp\rector\cache\test_3de7221bca6c581e090e825fb44bc1ab\ContainerSh7YppC\Rector_Core_HttpKernel_RectorKernelTest_3de7221bca6c581e090e825fb44bc1abDebugContainer.php:8774
C:\xampp7.3\htdocs\rector-src\tmp\rector\cache\test_3de7221bca6c581e090e825fb44bc1ab\ContainerSh7YppC\Rector_Core_HttpKernel_RectorKernelTest_3de7221bca6c581e090e825fb44bc1abDebugContainer.php:3046
C:\xampp7.3\htdocs\rector-src\tmp\rector\cache\test_3de7221bca6c581e090e825fb44bc1ab\ContainerSh7YppC\Rector_Core_HttpKernel_RectorKernelTest_3de7221bca6c581e090e825fb44bc1abDebugContainer.php:2346
C:\xampp7.3\htdocs\rector-src\tmp\rector\cache\test_3de7221bca6c581e090e825fb44bc1ab\ContainerSh7YppC\Rector_Core_HttpKernel_RectorKernelTest_3de7221bca6c581e090e825fb44bc1abDebugContainer.php:2356
C:\xampp7.3\htdocs\rector-src\tmp\rector\cache\test_3de7221bca6c581e090e825fb44bc1ab\ContainerSh7YppC\Rector_Core_HttpKernel_RectorKernelTest_3de7221bca6c581e090e825fb44bc1abDebugContainer.php:2326
C:\xampp7.3\htdocs\rector-src\vendor\symfony\dependency-injection\Container.php:248
C:\xampp7.3\htdocs\rector-src\vendor\symfony\dependency-injection\Container.php:228
C:\xampp7.3\htdocs\rector-src\packages\Testing\PHPUnit\AbstractTestCase.php:59
C:\xampp7.3\htdocs\rector-src\packages\Testing\PHPUnit\AbstractRectorTestCase.php:52
```

-----


I also tried using a autoload file via `Option::BOOTSTRAP_FILES`, but this didn't work either:

```
$ vendor/bin/phpunit packages-tests/BetterPhpDocParser/PhpDocInlineHtml/TestInlineHtmlTest.php --debug
PHPUnit 9.5.4 by Sebastian Bergmann and contributors.

Test 'Rector\Tests\BetterPhpDocParser\PhpDocInlineHtml\TestInlineHtmlTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))' started
Test 'Rector\Tests\BetterPhpDocParser\PhpDocInlineHtml\TestInlineHtmlTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))' ended


Time: 00:01.807, Memory: 30.00 MB

There was 1 error:

1) Rector\Tests\BetterPhpDocParser\PhpDocInlineHtml\TestInlineHtmlTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
Rector\Core\Exception\ShouldNotHappenException: Rector rule "Rector\Tests\BetterPhpDocParser\PhpDocInlineHtml\InlineHtmlRector" not found, please verify that the rule exists

C:\xampp7.3\htdocs\rector-src\src\DependencyInjection\CompilerPass\VerifyRectorServiceExistsCompilerPass.php:28
C:\xampp7.3\htdocs\rector-src\vendor\symfony\dependency-injection\Compiler\Compiler.php:91
C:\xampp7.3\htdocs\rector-src\vendor\symfony\dependency-injection\ContainerBuilder.php:736
C:\xampp7.3\htdocs\rector-src\vendor\symfony\http-kernel\Kernel.php:541
C:\xampp7.3\htdocs\rector-src\vendor\symfony\http-kernel\Kernel.php:780
C:\xampp7.3\htdocs\rector-src\vendor\symfony\http-kernel\Kernel.php:121
C:\xampp7.3\htdocs\rector-src\packages\Testing\PHPUnit\AbstractTestCase.php:39
C:\xampp7.3\htdocs\rector-src\packages\Testing\PHPUnit\AbstractRectorTestCase.php:50

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.

```

Reply:
At first sight, the `Rector\Tests\BetterPhpDocParser\PhpDocInlineHtml\InlineHtmlRector` does not respect PSR-4 autoload here. Maybe adding "/Source/" would help.

Only the files in `Fixture` are loaded with static reflection, to keep `Source` contract strict as possible.

Reply:
> At first sight, the `Rector\Tests\BetterPhpDocParser\PhpDocInlineHtml\InlineHtmlRector` does not respect PSR-4 autoload here. Maybe adding "/Source/" would help.

aah good point. forgot to adjust the namespace after moving this class. fixing the namespace seem to have fixed the autoloading case.